feat(web): Skip duplicates

xCJPECKOVERx opened the pull request:
## Description

Adds the option to skip a duplicate review. This doesn't remove the duplicate, just moves onto the next one in the duplicates list. Resets on page reload.

Fixes #10402

## How Has This Been Tested?

Try skipping duplicates, ensure index doesn't go out of bounds.

<details><summary><h2>Screenshots (if appropriate)</h2></summary>

<!-- Images go below this line. -->
<img width="875" height="495" alt="image" src="https://github.com/user-attachments/assets/3809dcf9-a465-4e87-99b1-02bf646e70e2" />


</details>

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

xCJPECKOVERx committed:
- add skip button to duplicates-compare-control

xCJPECKOVERx committed:
- cleanup

xCJPECKOVERx commented on web/src/lib/components/utilities-page/duplicates/duplicates-compare-control.svelte:
```diff
@@ -95,6 +97,10 @@
   };
+
+  const handleSkip = () => {
+    onSkip();
```
Is it preferred to instead call onSkip directly from the component, vs this "handleX" function?

jrasm91 reviewed (commented):
I'd probably use next/previous verbiage. Also you could include the index as a query param, so it goes to the same one on page refresh. `?index=5`

xCJPECKOVERx commented:
> I'd probably use next/previous verbiage. Also you could include the index as a query param, so it goes to the same one on page refresh. `?index=5`

If allowing refreshes, I'll add first/last buttons as well so we can hop back to the beginning.

xCJPECKOVERx committed:
- change to next/previous
- move buttons to duplicates page, intead of compareControl
- add param based control/position

xCJPECKOVERx committed:
- remove index param on keep/dedupe all

xCJPECKOVERx committed:
- cleanup

xCJPECKOVERx committed:
- cleanup index corrections

xCJPECKOVERx committed:
- add left/right arrow keyboard shortcuts for previous/next
- cleanup

xCJPECKOVERx committed:
- cleanup

xCJPECKOVERx commented:
> If allowing refreshes, I'll add first/last buttons as well so we can hop back to the beginning.

Added these controls, and now the index is entirely controlled by the 'index' param, with safeguards against NaN and out of bounds. Performing actions (stack, delete, keep) navigates to the next appropriate index.
